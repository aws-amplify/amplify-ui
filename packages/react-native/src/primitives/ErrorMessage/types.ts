import {
  ImageStyle,
  StyleProp,
  TextStyle,
  ViewProps,
  ViewStyle,
} from 'react-native';

import { IconProps } from '../Icon';

export interface ErrorMessageProps extends ViewProps {
  iconStyle?: IconProps['style'];
  labelStyle?: StyleProp<TextStyle>;
  onDismiss?: () => void;
}

export interface ErrorMessageStyles {
  container?: ViewStyle;
  icon?: ImageStyle;
  label?: TextStyle;
}
