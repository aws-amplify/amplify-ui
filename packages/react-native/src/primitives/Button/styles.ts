import { StyleSheet } from 'react-native';

import { StrictTheme } from '../../theme';
import { ButtonStyles } from './types';

export const getThemedStyles = (theme: StrictTheme): ButtonStyles => {
  const { tokens, components } = theme;
  const {
    colors,
    fontSizes,
    fontWeights,
    opacities,
    space,
    radii,
    borderWidths,
  } = tokens;

  return StyleSheet.create({
    container: {
      alignItems: 'center',
      backgroundColor: colors.transparent,
      justifyContent: 'center',
      paddingVertical: space.small,
      paddingHorizontal: space.medium,
      borderRadius: radii.small,
      ...components?.button?.container,
    },
    containerDefault: {
      borderWidth: borderWidths.small,
      borderColor: colors.border.primary,
      ...components?.button?.containerDefault,
    },
    containerPrimary: {
      backgroundColor: colors.brand.primary[80],
      ...components?.button?.containerPrimary,
    },
    containerLink: {
      ...components?.button?.containerLink,
    },
    disabled: {
      opacity: opacities[60],
      ...components?.button?.disabled,
    },
    pressed: {
      opacity: opacities[60],
      ...components?.button?.pressed,
    },
    text: {
      textAlign: 'center',
      color: colors.font.primary,
      fontSize: fontSizes.medium,
      fontWeight: fontWeights.bold,
      ...components?.button?.text,
    },
    textPrimary: {
      color: colors.font.inverse,
      ...components?.button?.textPrimary,
    },
    textLink: {
      color: colors.font.interactive,
      ...components?.button?.textLink,
    },
  });
};
