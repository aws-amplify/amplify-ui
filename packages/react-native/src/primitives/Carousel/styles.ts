import { StyleSheet } from 'react-native';

import { CarouselStyles } from './types';

export const styles: CarouselStyles = StyleSheet.create({
  indicator: {
    bottom: 0,
    flexDirection: 'row',
    justifyContent: 'center',
    left: 0,
    position: 'absolute',
    right: 0,
  },
});
