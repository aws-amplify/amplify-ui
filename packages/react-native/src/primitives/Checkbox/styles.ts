import { StyleSheet } from 'react-native';

import { StrictTheme } from '../../theme';
import { CheckboxStyles } from './types';
import { getFlexDirectionFromLabelPosition } from '../Label/utils';
import { LabelPosition } from '../Label/types';

export const getThemedStyles = (
  theme: StrictTheme,
  labelPosition: LabelPosition
): CheckboxStyles => {
  const { tokens, components } = theme;
  const { opacities, space } = tokens;

  return StyleSheet.create({
    container: {
      alignItems: 'center',
      padding: space.xs,
      flexDirection: getFlexDirectionFromLabelPosition(labelPosition),
      ...components?.checkbox?.container,
    },
    disabled: {
      opacity: opacities[60],
      ...components?.checkbox?.disabled,
    },
    icon: {
      ...components?.checkbox?.icon,
    },
    label: {
      ...components?.checkbox?.label,
    },
  });
};
