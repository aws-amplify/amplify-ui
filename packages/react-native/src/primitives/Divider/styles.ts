import { StyleSheet } from 'react-native';

import { StrictTheme } from '../../theme';
import { DividerStyles } from './types';

export const getThemedStyles = (theme: StrictTheme): DividerStyles => {
  const { tokens, components } = theme;
  const { colors, fontSizes, space } = tokens;

  return StyleSheet.create({
    container: {
      alignItems: 'center',
      flexDirection: 'row',
      ...components?.divider?.container,
    },
    label: {
      fontSize: fontSizes.medium,
      margin: space.medium,
      textAlign: 'center',
      ...components?.divider?.label,
    },
    line: {
      backgroundColor: colors.black,
      flex: 1,
      height: StyleSheet.hairlineWidth,
      ...components?.divider?.line,
    },
  });
};
