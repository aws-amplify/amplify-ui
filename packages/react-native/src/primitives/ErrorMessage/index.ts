export { default as ErrorMessage } from './ErrorMessage';
export { ErrorMessageProps, ErrorMessageStyles } from './types';
