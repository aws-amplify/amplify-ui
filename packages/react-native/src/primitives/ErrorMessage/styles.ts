import { StyleSheet } from 'react-native';

import { StrictTheme } from '../../theme';
import { ErrorMessageStyles } from './types';

export const getThemedStyles = (theme: StrictTheme): ErrorMessageStyles => {
  const {
    tokens: { colors, fontSizes, space },
    components,
  } = theme;

  return StyleSheet.create({
    container: {
      alignItems: 'center',
      backgroundColor: colors.background.error,
      flexDirection: 'row',
      paddingHorizontal: space.xs,
      paddingVertical: space.small,
      ...components?.errorMessage?.container,
    },
    icon: {
      margin: space.xs,
      tintColor: colors.font.error,
      ...components?.errorMessage?.icon,
    },
    label: {
      color: colors.font.error,
      flex: 1,
      fontSize: fontSizes.medium,
      paddingHorizontal: space.xs,
      ...components?.errorMessage?.label,
    },
  });
};
