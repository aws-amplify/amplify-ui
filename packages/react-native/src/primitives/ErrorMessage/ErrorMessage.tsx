import React from 'react';
import { Text, View } from 'react-native';

import { icons } from '../../assets';
import { Icon } from '../Icon';
import { IconButton } from '../IconButton';

import { useTheme } from '../../theme';
import { getThemedStyles } from './styles';
import { ErrorMessageProps } from './types';

export const CLOSE_BUTTON_TEST_ID =
  'authenticator--error-message--close-button';

export default function ErrorMessage({
  children,
  iconStyle,
  labelStyle,
  onDismiss,
  style,
  ...rest
}: ErrorMessageProps): JSX.Element {
  const theme = useTheme();
  const themedStyle = getThemedStyles(theme);

  return (
    <View
      {...rest}
      accessibilityRole="alert"
      style={[themedStyle.container, style]}
    >
      <Icon
        size={20}
        source={icons.error}
        style={[themedStyle.icon, iconStyle]}
      />
      <Text style={[themedStyle.label, labelStyle]}>{children}</Text>
      {onDismiss ? (
        <IconButton
          onPress={onDismiss}
          size={20}
          source={icons.close}
          style={themedStyle.icon}
          testID={CLOSE_BUTTON_TEST_ID}
        />
      ) : null}
    </View>
  );
}
