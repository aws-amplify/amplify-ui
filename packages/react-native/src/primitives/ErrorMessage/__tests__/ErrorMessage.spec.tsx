import React from 'react';
import { fireEvent, render, renderHook } from '@testing-library/react-native';

import { useTheme } from '../../../theme';
import { CLOSE_BUTTON_TEST_ID } from '../ErrorMessage';
import ErrorMessage from '../ErrorMessage';
import { getThemedStyles } from '../styles';

describe('ErrorMessage', () => {
  it('renders default ErrorMessage as expected', () => {
    const { toJSON, queryByTestId } = render(
      <ErrorMessage>Default ErrorMessage</ErrorMessage>
    );
    expect(toJSON()).toMatchSnapshot();
    expect(queryByTestId(CLOSE_BUTTON_TEST_ID)).toBeFalsy();
  });

  it('handles an onDismiss callback', () => {
    const customOnDismiss = jest.fn();
    const { getByTestId } = render(
      <ErrorMessage onDismiss={customOnDismiss}>
        Default ErrorMessage
      </ErrorMessage>
    );

    const dismissButton = getByTestId(CLOSE_BUTTON_TEST_ID);
    fireEvent.press(dismissButton);

    expect(customOnDismiss).toHaveBeenCalledTimes(1);
  });

  it('applies theme and style props', () => {
    const customStyle = { backgroundColor: 'blue' };
    const customLabelStyle = { color: 'white' };
    const labelText = 'Themed ErrorMessage';

    const { getByRole, getByText, toJSON } = render(
      <ErrorMessage labelStyle={customLabelStyle} style={customStyle}>
        {labelText}
      </ErrorMessage>
    );

    const { result } = renderHook(() => useTheme());
    const themedStyle = getThemedStyles(result.current);

    const alert = getByRole('alert');
    const label = getByText(labelText);

    expect(alert.props.style).toStrictEqual([
      themedStyle.container,
      customStyle,
    ]);

    expect(label.props.style).toStrictEqual([
      themedStyle.label,
      customLabelStyle,
    ]);
    expect(toJSON()).toMatchSnapshot();
  });
});
