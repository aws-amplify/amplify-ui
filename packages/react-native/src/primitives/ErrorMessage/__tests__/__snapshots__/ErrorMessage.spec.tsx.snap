// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage applies theme and style props 1`] = `
<View
  accessibilityRole="alert"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "hsl(0, 75%, 85%)",
        "flexDirection": "row",
        "paddingHorizontal": 8,
        "paddingVertical": 12,
      },
      Object {
        "backgroundColor": "blue",
      },
    ]
  }
>
  <Image
    accessibilityRole="image"
    source={
      Object {
        "testUri": "../../../src/assets/icons/error.png",
      }
    }
    style={
      Array [
        Object {
          "height": 20,
          "resizeMode": "contain",
          "tintColor": undefined,
          "width": 20,
        },
        Array [
          Object {
            "margin": 8,
            "tintColor": "hsl(0, 100%, 20%)",
          },
          undefined,
        ],
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "hsl(0, 100%, 20%)",
          "flex": 1,
          "fontSize": 16,
          "paddingHorizontal": 8,
        },
        Object {
          "color": "white",
        },
      ]
    }
  >
    Themed ErrorMessage
  </Text>
</View>
`;

exports[`ErrorMessage renders default ErrorMessage as expected 1`] = `
<View
  accessibilityRole="alert"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "hsl(0, 75%, 85%)",
        "flexDirection": "row",
        "paddingHorizontal": 8,
        "paddingVertical": 12,
      },
      undefined,
    ]
  }
>
  <Image
    accessibilityRole="image"
    source={
      Object {
        "testUri": "../../../src/assets/icons/error.png",
      }
    }
    style={
      Array [
        Object {
          "height": 20,
          "resizeMode": "contain",
          "tintColor": undefined,
          "width": 20,
        },
        Array [
          Object {
            "margin": 8,
            "tintColor": "hsl(0, 100%, 20%)",
          },
          undefined,
        ],
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "hsl(0, 100%, 20%)",
          "flex": 1,
          "fontSize": 16,
          "paddingHorizontal": 8,
        },
        undefined,
      ]
    }
  >
    Default ErrorMessage
  </Text>
</View>
`;
